Fix Darwin config use spectrum with ppc64le

JoshuaSBrown opened the pull request:
<!--Provide a general summary of your changes in the title above, for
example "Add AMR unit test for cell centered fields.".  Please avoid
non-descriptive titles such as "Addresses issue #8576".-->

## PR Summary

<!--Please provide at least 1-2 sentences describing the pull request in
detail.  Why is this change required?  What problem does it solve?-->
This pr fixes the default build configuration on Darwin. Openmpi compatibility with running gpus was causing problem spectrum mpi does not appear to have the same problems. Namely the advection convergence and performance tests were failing. 
<!--If it fixes an open issue, please link to the issue here.-->

## PR Checklist

<!-- Note that some of these check boxes may not apply to all pull requests -->

- [x] Code passes cpplint
- [x] New features are documented.
- [x] Code is formatted
- [x] Changes are summarized in CHANGELOG.md
- [x] (@lanl.gov employees) Update copyright on changed files

Commit:
Fix Darwin config use spectrum with ppc64le

ghost commented:
<img src="https://www.deepcode.ai/icons/green_check.svg" width= "50px" align= "left"/> Congratulations :tada:. DeepCode [analyzed](https://www.deepcode.ai/app/gh/lanl/parthenon/41a8923c1c12141d3ac7ff4955f9c7c55a90bc3c/lanl/parthenon/1b5c21a8d77a61e15d3d60a43aa36a3a391d4cb8/pr/_/%2F/code/?utm_source=gh_review&c=0&w=0&i=0&) your code in 0.848 seconds and we found no issues. Enjoy a moment of no bugs :sunny:.

#### 👉 View analysis in [**DeepCode’s Dashboard**](https://www.deepcode.ai/app/gh/lanl/parthenon/41a8923c1c12141d3ac7ff4955f9c7c55a90bc3c/lanl/parthenon/1b5c21a8d77a61e15d3d60a43aa36a3a391d4cb8/pr/_/%2F/code/?utm_source=gh_review&c=0&w=0&i=0&) | [_Configure the bot_](https://www.deepcode.ai/app/gh/?ownerconfig=lanl)

Commit:
Add Changelog

JoshuaSBrown commented:
This should be good to go I've tested locally in my user account, but I'm still installing all the spack modules in the parthenon project space so should wait till I test them before this is merged.

AndrewGaspar commented on cmake/machinecfg/Darwin.cmake:
```diff
@@ -232,9 +237,14 @@ endif()
+    set(MPI_ROOT 
+    /projects/opt/${DARWIN_ARCH}${DARWIN_MICROARCH_PATH}/openmpi/${DARWIN_MPI_VERSION}-gcc_${DARWIN_${DARWIN_COMPILER}_VERSION}
+        CACHE STRING "MPI Location")
+  else()
```
```suggestion
  elseif(DARWIN_MPI_PACKAGE STREQUAL "smpi")
```

AndrewGaspar commented on cmake/machinecfg/Darwin.cmake:
```diff
@@ -232,9 +237,14 @@ endif()
 if (DARWIN_COMPILER MATCHES "GCC")
-    set(MPI_ROOT
-        /projects/opt/${DARWIN_ARCH}${DARWIN_MICROARCH_PATH}/openmpi/${DARWIN_MPI_VERSION}-gcc_${DARWIN_${DARWIN_COMPILER}_VERSION}
+  if(DARWIN_MPI_PACKAGE MATCHES "openmpi")
```
```suggestion
  if(DARWIN_MPI_PACKAGE STREQUAL "openmpi")
```

AndrewGaspar commented on cmake/TestSetup.cmake:
```diff
@@ -44,6 +44,55 @@ function(record_driver arg)
+    list(APPEND TMPARGS "${TEST_MPIEXEC}")
+  # use CMake determined mpiexec
+  else()
+    message("MPI executable ${MPIEXEC_EXECUTABLE}")
```
This looks spammy.

```suggestion
```

Commit:
Remove comment

Commit:
Apply suggestions from code review

I can now confirm that the project space on Darwin is working. This is good to go pending further review.

Co-authored-by: Andrew Gaspar <agaspar@lanl.gov>

Commit:
Merge branch 'develop' into JoshuaSBrown/fix-darwin-build-config

JoshuaSBrown commented:
@AndrewGaspar

JoshuaSBrown commented:
https://re-git.lanl.gov/eap-oss/parthenon/-/jobs/97386 so this shows the ci now passing on Darwin, which it hasn't been. :)

JoshuaSBrown commented:
@AndrewGaspar @jlippuner @Yurlungur @pgrete ping ping. :)

AndrewGaspar commented:
> ping ping. :)

You should get that looked at.

Yurlungur reviewed (approved):
Thanks for fixing this.
